src/services/importService.ts

```diff
@@ -0,0 +1,191 @@
+import shortid from "shortid";
+import { IProject, ITag, IConnection, AppError, ErrorCode,
+        IAssetMetadata, IRegion, RegionType, AssetState, IFileInfo } from "../models/applicationState";
+import { IV1Project } from "../models/v1Models";
+import { AssetService } from "./assetService";
+import { randomIntInRange } from "../common/utils";
+import TagColors from "../react/components/common/tagsInput/tagColors.json";
+
+/**
+ * Functions required for an import service
+ * @member convertV1 - Converts a v1 project to v2 project
+ * @member generateAssets - Generates v2 assets based on v1 project file
+ * @member generateConnections - Generates v2 connections based on location of v1 project file
+ */
+export interface IImportService {
+    convertV1(project: IFileInfo): Promise<IProject>;
+}
+
+/**
+ * @name - Import Service
+ * @description - Functions for importing v1 projects to v2 application
+ */
+export default class ImportService implements IImportService {
+    public async convertV1(project: IFileInfo): Promise<IProject> {
+        let originalProject: IV1Project;
+        let convertedProject: IProject;
+        let connections: IConnection[];
+        let parsedTags: ITag[];
+        let generatedAssetMetadata: any;
+
+        try {
+            originalProject = JSON.parse(project.content as string);
+        } catch (e) {
+            throw new AppError(ErrorCode.ProjectInvalidJson, "Error parsing JSON");
+        }
+
+        parsedTags = this.parseTags(originalProject);
+
+        connections = this.generateConnections(project);
+
+        // map v1 values to v2 values
+        convertedProject = {
+            id: shortid.generate(),
+            name: project.file.name.split(".")[0],
+            version: "v1-to-v2",
+            securityToken: `${project.file.name.split(".")[0]} Token`,
+            description: "Converted V1 Project",
+            tags: parsedTags,
+            sourceConnection: connections[0],
+            targetConnection: connections[1],
+            exportFormat: null,
+            videoSettings: {
+                frameExtractionRate: 15,
+            },
+            autoSave: true,
+            assets: {},
+        };
+
+        const assetService = new AssetService(convertedProject);
+        generatedAssetMetadata = this.generateAssets(project, assetService);
+
+        const saveAssets = generatedAssetMetadata.map((assetMetadata) => {
+            assetMetadata.then((metadata) => {
+                return assetService.save(metadata);
+            });
+        });
+
+        try {
+            Promise.all(saveAssets);
+        } catch (e) {
+            throw e;
+        }
+
+        return convertedProject;
+    }
+
+    /**
+     * Generate connections from v1 project file location
+     * @param project - V1 Project Content and File Information
+     */
+    private generateConnections(project: any): IConnection[] {
```

Comment:
As mentioned above, let's reuse the same connection for both the source and target connections... they can always go back and change if needed.